workflow/controller/cache/configmap_cache.go

```diff
@@ -102,10 +118,14 @@ func (c *configMapCache) Save(ctx context.Context, key string, nodeId string, va
 		}
 	}
 
+	creationTime := time.Now()
+	cache.SetLabels(map[string]string{common.LabelKeyCacheLastHitTimestamp: creationTime.Format(time.RFC3339)})
+
 	newEntry := Entry{
 		NodeID:            nodeId,
 		Outputs:           value,
-		CreationTimestamp: metav1.Time{Time: time.Now()},
+		CreationTimestamp: metav1.Time{Time: creationTime},
```

Comment:
I wonder what the utility of creation timestamp is now that we introduce last hit timestamp... perhaps maxAge should always reference lastHitTimestamp instead, although this would break some user expecations about how maxAge works....

What do you think?

Reply:
This is only when it's firstly saved. When the entry is actually hit/loaded every time, the `LastHitTimestamp` would change. I think `maxAge` should stick with `CreationTimestamp` as the expectation for age is after something is initially born/created. Interesting analysis can be done by utilizing both timestamps, e.g. the cache is very old but still being used so maybe it's time to delete/refresh it?

Reply:
> Interesting analysis can be done by utilizing both timestamps, e.g. the cache is very old but still being used so maybe it's time to delete/refresh it?

Yeah I agree, this is good potential. Out of scope for this PR though